### ISSUE TITLE
VSCode 1.62.0 Session contents restored from (current date) powershell

### ISSUE THREAD

**Issue by christalk** [microsoft/vscode#136566]
- VS Code Version:  1.62.0
- OS Version: Windows10  21H1   
![VSCode  1 62 0](https://user-images.githubusercontent.com/26896253/140604760-ca06147d-30a7-48db-b689-54cc984f6b1e.png)
![VsCode 1 62 0](https://user-images.githubusercontent.com/26896253/140604761-917a34b9-c9f1-46ba-a1d4-9814aef57131.png)

Steps to Reproduce: (same with Ubuntu , but "bash" insteed of "powershell"

At each new VSCode session, the terminal window displays the prompt but only after several seconds.
2 - Then there is also the sentence:
Session contents restored from (today's date)
3 - and sometimes more than one powershell line on the right

**Comment by gjsjohnmurray** [comment]
This is as designed.

The "Session contents restored..." line tells you that everything above here was preserved when you previously closed your VS Code window, and was reloaded when you launched VS Code again.

The two "powershell" items on the right of your screenshot are tabs representing two terminals that you have started, perhaps by using the "+" button above them. Hover over these tabs to discover what you can do with them. Or drag them.

See https://code.visualstudio.com/docs/editor/integrated-terminal for more information.

**Comment by christalk** [comment]
Thanks.
Regards.

**Comment by DesertBear** [comment]
This is not wanted behavior and is also buggy.

I use Visual Studio Code purely as a replacement for the PowerShell ISE and therefore it's used only for PowerShell.  VSCode 1.61 was fine, however the update to VSCode 1.62 is problematic.

**Background:**
VSCode is set so that when opening, a new editor is opened, which is automatically set to PowerShell.  The terminal window is also set to automatically open the terminal with the "PowerShell Integrated Terminal" (_Integrated Console Show on Startup_). This is a PowerShell ISE type environment and is therefore ordinarily instantly ready to go as soon as I open VSCode.  This is no longer the case with 1.62.

I also use the "PowerShell Integrated Terminal" instead of PWSH, because when using PWSH, the editor passes the PowerShell code from the editor to the terminal line-by-line, whereas with the "PowerShell Integrated Terminal" it works correctly.  In addition, when using F5 (Run) or F8 (Run Selection) to run the PowerShell code that's in the editor, it will automatically run in "PowerShell Integrated Terminal" anyway.  Therefore, I do not ever want to be switching between PWSH and "PowerShell Integrated Terminal" – I want to be using "PowerShell Integrated Terminal" all of the time.

Since 1.62, it no longer works correctly.

**Problem 1:**
When opening VSCode, the terminal window appears for the first 6-7 seconds until PowerShell loads, then the terminal window dissapears.  Everytime, I need to press [CTRL + '] to get it back again.

**Problem 2:**
The VSCode terminal is set to open the "PowerShell Integrated Terminal" on startup [_Extensions > Powershell Configuration > Integrated Console Show on Startup_].  However, it now opens PWSH instead – which I do not want.

**Problem 3:**
When VSCode opens PWSH instead of the "PowerShell Integrated Terminal", it says "_Session contents restored from 06/11/2021 at 13:51:09_".  This is what the OP reported.  However, I do not want anything restored in the terminal.

**Problem 4:**
The PWSH window that opens instead of the "PowerShell Integrated Terminal" isn't usable.  I.E.

**a)** It's not possible to type anything in the PWSH terminal.  All you have it a blinking cursor, but are not able to type anything.

**b)** Quite often it's not possible to delete the PWSH instance by pressing the dustbin/trash icon in the top-right of the terminal window.  You need to try typing in the terminal window first (even though it won't allow you to type anything) before the delete icon will start working.

**c)** Sometimes you end up with multiple PWSH sessions in the list, despite not even wanting one.

**Video 1** 
VSCode 1.62 (Bad).mp4 (shows the above problems).
https://user-images.githubusercontent.com/27423064/140616061-23b91ac9-02a7-49de-bdc6-ee4ce28fb043.mp4

**Video 2**
VSCode 1.61 (Good).mp4 (shows it working correctly as it was before).
https://user-images.githubusercontent.com/27423064/140616087-b3929107-f82b-4f13-a619-ae78333f9081.mp4

**VSCode settings:**
See attached ZIP file for VSCode settings file.
[settings.zip](https://github.com/microsoft/vscode/files/7490823/settings.zip)


-----
Windows 10 - 21H1 (19043.1288)
PowerShell - 7.1.5
VSCode - 1.62.0
VSCode PowerShell extension - 2021.10.2

**Comment by gjsjohnmurray** [comment]
@DesertBear did you try changing the setting documented at https://code.visualstudio.com/docs/editor/integrated-terminal#_terminal-process-reconnection on the page I pointed @christalk to?

**Comment by DesertBear** [comment]
Disabling that setting works and VSCode reverts to the pre 1.62 behaviour.  Thanks.

`"terminal.integrated.persistentSessionReviveProcess": "never"`


Disabling the [below setting](https://code.visualstudio.com/updates/v1_54#_retain-terminal-processes-on-window-reload) also works:

`"terminal.integrated.enablePersistentSessions": false`

**Comment by smry-web** [comment]
I understand it is as designed - but _why_ was it designed like that I guess is my question

**Comment by jntSeth** [comment]
> Disabling that setting works and VSCode reverts to the pre 1.62 behaviour. Thanks.
> 
> `"terminal.integrated.persistentSessionReviveProcess": "never"`
> 
> Disabling the below setting [from VSCode 1.54](https://code.visualstudio.com/updates/v1_54#_retain-terminal-processes-on-window-reload) also works:
> 
> `"terminal.integrated.enablePersistentSessions": false`

Thank you for this. I agree, this change is absolutely awful. I just wasted 30 minutes because VSCode refused to release my PnPContext.

**Comment by jntSeth** [comment]
> The described behavior is how it is expected to work. If you disagree, please explain what is expected and what is not in more detail. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.
> 
> Happy Coding!

Whatever this change is, it no longer requires AUTHENTICATION. A previous set of credentials seems to be stored somewhere as I'm not being asked for any authentication for the SharePoint online site in question, even though all users have MFA enabled.

I have killed every running terminal, in VSCode, I have killed its process, I have attempted to exist every pssession. I have restarted VSCode multiple times, and if I request the site I previously authenticated for.... it just lets me  in.

This is very bad.

**Comment by BlackV** [comment]
> 
> 
> > Disabling that setting works and VSCode reverts to the pre 1.62 behaviour. Thanks.
> > `"terminal.integrated.persistentSessionReviveProcess": "never"`
> > Disabling the below setting [from VSCode 1.54](https://code.visualstudio.com/updates/v1_54#_retain-terminal-processes-on-window-reload) also works:
> > `"terminal.integrated.enablePersistentSessions": false`
> 
> Thank you for this. I agree, this change is absolutely awful. I just wasted 30 minutes because VSCode refused to release my PnPContext.

can confirm 100% awful

**Comment by LeGEC** [comment]
I see a mention to a new `terminal.integrated.persistentSessionReviveProcess` setting in the release notes of [version 1.61 (September 2021)](https://code.visualstudio.com/updates/v1_61#_restore-terminal-sessions-across-application-restarts) :

> In this release, there is a new opt-in setting for recording and restoring the buffer and using it to recreate processes. Set `terminal.integrated.persistentSessionReviveProcess` to enable this.

I don't see a note stating that this option was turned to an opt-out setting.

Did I miss that part ?

**Comment by viniciusvillas** [comment]
This feature would be great if it really worked at all but, as reported by @DesertBear, it's way too broken for PowerShell. It's unusable in the current state, and really should've been treated as a beta opt-in feature instead of an opt-out feature.
Disabling the "persistent session" restores the former usable condition of VS Code for PowerShell. So, no drama there.

**Comment by christalk** [comment]
Thank you for this..

**Comment by erwinagpasa** [comment]
<img width="922" alt="Screen Shot 2021-11-19 at 6 22 10 AM" src="https://user-images.githubusercontent.com/637956/142559553-3089959d-c993-4a25-8026-0e45727efe5f.png">
``

**Comment by SoulSeekkor** [comment]
Just adding to this that the change that made this happen was god awful and made me despise whenever I had to use VS Code for Powershell scripts... Every single time I opened VS Code the terminal panel would close almost immediately which looked very much like buggy behavior.  To reinforce this, going to the menu to show it again had "restored session contents" in the window which looks like something failed to close properly on the last session, with extra terminals listed to the right.  Every time VS Code was opened it was creating new ones I had to delete over and over, very frustrating.

Thank you to the person above that commented on the horrid revive setting to make this behave properly again!
